src/features/commands/report.js

```diff
@@ -0,0 +1,340 @@
+const { SlashCommandBuilder } = require('@discordjs/builders');
+const { MessageActionRow, MessageButton, MessageSelectMenu, MessageSelectOptionData, Constants, MessageEmbed } = require('discord.js');
+const { report } = require('process');
+const { promisify } = require('util')
+const Report = require('../../reportObject.js');
+const UserCases = require('../../userCases.js');
+const db = require('../../db');
+const { fetchRules } = require('./../../api/discord-gating');
+const { values } = require('lodash');
+const wait = promisify(setTimeout);
+const MessageReports = require('../../reportTypes/message.js');
+
+function getMessageId(linkOrId)
+{
+    if (!isNaN(linkOrId)) {
+        return linkOrId;
+    } else {
+        const match = linkOrId.match('/discord.com/channels/([0-9]+)/([0-9]+)/([0-9]+)');
+        return match[3];
+    }
+}
+
+// Minimum time before allowing the user to report
+const reportAwaitTime = 5000;
+// Time before dropping the report. This is used to avoid users create many interactions that the bot collects.
+const reportTimeout = 30000;
+
+const alreadyReportedText = 'You already reported this message.';
+const alreadyReviewedText = 'This report was already reviewed by a moderator.';
+const blockedReportingText = `You are prevented from reporting, this could be because you abused the system.\n
+If you want more info, or if you think this is an error, contact moderators.`;
+const warnText = 'You are about to report a message. Abusing the report system coud lead to punishment. If this is really your intent you can go ahead.';
+const genericReportText = 'Moderators will now review your report. Please note that they are not robots: it might take some time before an action is taken. You will receive a DM once moderators review your report, make sure you have enabled them for this server!';
```

Comment:
```suggestion
const genericReportText = 'Report is pending a review, please note that this might take some time. You will receive a direct message once moderators review your report, make sure you have enabled them for this server!';
```